components/input/src/input-field/input-field.js

```diff
@@ -66,7 +66,13 @@ class InputField extends React.Component {
                         onChange={onChange}
                         name={name}
                         type={type}
-                        value={value || ''}
+                        value={
+                            value === null ||
+                            value === undefined ||
+                            Number.isNaN(value)
+                                ? ''
+                                : value
+                        }
```

Comment:
Could this just use a [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)?  Are we also trying to handle `NaN` [falsy values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) or just trying to make sure `0` can be passed?

```suggestion
                        value={value ?? ''}
```

Reply:
Yes, i was not sure here. I doubt NaN is ever going to be passed as a value, but I thought maybe safer to keep the previous behaviour for everything that is not 0, -0 or other forms of 0? But happy to change it as well. Do you think `??` would be better?